### PR TITLE
Oauth lito

### PR DESCRIPTION
**Pull request by ljwhite**
# Description
 - A user can login with Google Oauth
 - The user is prompted to allow the application to manage books on Google
 - Upon logging in, a user model is created with the user information from Google
 - The user stays is provided a session to stay logged in 
 - The user can subsequently log out on the welcome page. This will delete the session

# Closes issue(s)
 - Google Oauth (#4)

# How to test / reproduce
 - Can test on rails server 
 - RSpec feature tests for logging in and logging out
 - RSpec model test for user
 - There is a support subdirectory in spec that has a login_as_user method. This way, you'll only need to call this method to mock the login process
 
# Screenshots
Screen Shot 2020-09-11 at 4 02 47 PM

# Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist
- [x] I have tested this code
- [x] I have updated the Readme

# Other comments
Rubocop is providing some interesting suggestions. We will need a future spike to address.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.update_or_create(request.env['omniauth.auth'])
+    session[:id] = user.id
+    redirect_to root_path
+  end
+
+  def destroy
+    session.clear
```
Comment (Strompy) [inline app/controllers/sessions_controller.rb:9]: So easy!

File: spec/support/session_helpers.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+module SessionHelpers
+
+  def stub_omniauth
```
Comment (Strompy) [inline spec/support/session_helpers.rb:3]: This is a cool way to dry up tests

**Review by Strompy — APPROVED** [review]

File: spec/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    new_user = User.first
+
+    expect(new_user.provider).to eq('google')
+    expect(new_user.uid).to eq('12345678910')
+    expect(new_user.email).to eq('lito@litowhite.com')
+    expect(new_user.first_name).to eq('Lito')
+    expect(new_user.last_name).to eq('White')
+    expect(new_user.token).to eq('abcdefg12345')
+    expect(new_user.refresh_token).to eq('12345abcdefg')
+    expect(new_user.oauth_expires_at).to eq(auth[:credentials][:expires_at])
```
Comment (ajtran303) [inline spec/models/user.rb:29]: I think for the prior assertions 22-28, referencing the hash like in like 29 would make the spec more resilient.

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -25,7 +25,7 @@
 # directory. Alternatively, in the individual `*_spec.rb` files, manually
 # require only the support files necessary.
 #
-# Dir[Rails.root.join('spec', 'support', '**', '*.rb')].sort.each { |f| require f }
+Dir[Rails.root.join('spec', 'support', '**', '*.rb')].sort.each { |f| require f }
```
Comment (ajtran303) [inline spec/rails_helper.rb:28]: So I read the comment block above, and it seems that this line may impact how fast our specs run.

**Review by ajtran303 — COMMENTED** [review]
Thanks Lito. Your feature specs are awesome -- there's a lot going on under the hood with the session helper module to make that possible.

File: db/schema.rb [hunk 5]
```diff
@@ -10,9 +10,22 @@
...
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "users", force: :cascade do |t|
+    t.string "first_name"
+    t.string "last_name"
+    t.string "email"
+    t.string "token"
+    t.string "refresh_token"
+    t.datetime "oauth_expires_at"
```
[the remark this answers is not in the document]
Comment (ljwhite) [reply db/schema.rb:24]: I'm not sure I understand the question?

File: spec/rails_helper.rb (hunk 4 above)
Comment (ljwhite) [reply spec/rails_helper.rb:28]: Yeah, it might. I guess the argument is that we'll probably need the login helper spec for pretty much every test

**Pull request merged**